bears/jinja2/Jinja2Bear.py

```diff
@@ -0,0 +1,370 @@
+
+    replacement = "{before}{label}{after}".format(
+        before=content_before,
+        after=content_after,
+        label=expected_label
+    )
+    diff.change_line(
+        line_number,
+        line,
+        replacement)
+    return {filename: diff}
+
+
+def has_required_spacing(string, required_spacing):
+    """
+    Checks if a string has the required amount of spaces
+    on each side.
+
+    :param string:           The string which is to be checked.
+    :param required_spacing: The number of spaces expected on both
+                             sides of the string.
+    :return:                 True if the string has the required
+                             number of spaces on both sides, False
+                             otherwise.
+
+    >>> has_required_spacing("  foo  ", 2)
+    True
+    >>> has_required_spacing(" foo", 1)
+    False
+    """
+    leading_spaces = len(string) - len(string.lstrip(' '))
+    trailing_spaces = len(string) - len(string.rstrip(' '))
+
+    return (
+        leading_spaces == required_spacing
+        and trailing_spaces == required_spacing)
+
+
+class Jinja2Bear(LocalBear):
+    LANGUAGES = {"Jinja2"}
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+    ASCIINEMA_URL = 'https://asciinema.org/a/azi6u1gcxutoxn0l7xpu4pljp'
+    CAN_DETECT = {'Syntax'}
+    CAN_FIX = {'Formatting', 'Documentation'}
+
+    VARIABLE_REGEX = re.compile(
+        r"(?P<open>{{)(?P<content>.*?)(?P<close>}})")
+    CONTROL_START_REGEX = re.compile(
+        r"(?P<open>{%)(?P<content>\s*(for|if).*?)(?P<close>%})")
+    CONTROL_END_REGEX = re.compile(
+        r"(?P<open>{%)(?P<content>\s*end(for|if)\s*?)"
+        r"(?P<close>%})(?P<label>{#.*?#})?")
+
+    def handle_control_spacing_issue(self, file, filename, line, line_number,
+                                     control_spacing, match_object):
+        """
+        Handles incorrectly spaced control tags (if and for),
+        both start and end tags
+
+        :param control_spacing: The number of spaces placed around control tags
+                                as required by the settings.
+        :param match_object:    A Match object containing the groups ``open``,
+                                ``close`` containing the opening and closing
+                                delimiters of a Jinja2 tag and ``content``
+                                containing everything in between.
+        """
+        diff = generate_spacing_diff(
+            file, filename, line, line_number, match_object, control_spacing)
+        return Result.from_values(
+            origin=self,
+            message="Control blocks should be spaced with "
+                    "`{0}` spaces on each side.".format(
+                        control_spacing),
+            file=filename,
+            line=line_number,
+            column=match_object.start(1) + 1,
+            end_line=line_number + 1,
```

Comment:
why `line_number + 1` and not just `line_number`? The match is all in the smae line right?